### ISSUE TITLE
Realm 1.6.0 conflict with dagger

### ISSUE THREAD

**Issue by DmytroTols** [realm/realm-kotlin#1252]
I have a module(only Realm 1.6.0 implemented) with my Database and RealmObjects,my main program has dagger and implementation of module,with this I no longer can create global variables like:
```
 class MainActivity : AppCompatActivity() {
    var dog: Dog= Dog()  

    override fun onCreate(savedInstanceState: Bundle?) {
        ....
    }
}
```
it will raise error:
```
error: cannot access Dog
    private gr.mastersdk.database.model.Dog dog;

undeclared type variable: T
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```
But the same code will work if I create this entity inside function like:
```
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var dog = Dog()
    }
}
```
Not really sure why this conflict exists,but having find any way out of it
dagger that I'm using: 'com.google.dagger:hilt-android:2.41'

**Comment by rorbech** [comment]
Hi @DmytroTols. Thanks for the report. I can reproduce your observation in a sample project, but will have to investigate the cause of the issue.

**Comment by DmytroTols** [comment]
Sure,thank you for quick response

**Comment by rorbech** [comment]
This only seems to be an issue with incremental builds. If I add the field it fails, but after a clean build it works and continues to do so until next time the properties that references a `RealmObject` is changed. If I change the source of the referenced class, `Doc` in the same go then it works. Could be something with the classes not being available on the class path of the kapt task when doing incremental builds. I don't know why this differentiates for classes implementing `RealmObject` since we transform them inline ... but the output might be placed in a different build intermediate build dir  which is somehow now included in the kapt build path 👀

**Comment by rorbech** [comment]
Seems to be a more general issue as I was able just provoke the same error without hilt in #1256

**Comment by rorbech** [comment]
The fix is now available in `1.6.1`.
